tests/model_fields/test_charfield.py

```diff
@@ -44,6 +44,10 @@ def test_charfield_with_choices_cleans_valid_choice(self):
         f = models.CharField(max_length=1, choices=[('a', 'A'), ('b', 'B')])
         self.assertEqual('a', f.clean('a', None))
 
+    def test_charfield_with_choices_cleans_valid_int_choice(self):
+        f = models.CharField(max_length=1, choices=[(1, 'One'), (2, 'Two')])
```

Comment:
`CharField` should only accept strings, not integers. This should be `choices=[('1', 'One'), ('2', 'Two')]` and then there is no issue.